mitmproxy/tools/web/app.py

```diff
@@ -308,14 +310,19 @@ class DumpFlows(RequestHandler):
     def get(self):
         self.set_header("Content-Disposition", "attachment; filename=flows")
         self.set_header("Content-Type", "application/octet-stream")
-
-        bio = BytesIO()
-        fw = io.FlowWriter(bio)
-        for f in self.view:
-            fw.add(f)
-
-        self.write(bio.getvalue())
-        bio.close()
+        self.set_header("Content-Encoding", "gzip")
```

Comment:
Removed. I guess this would be a case for another PR anyways, doing it for all, if Accept-Encoding contains gzip, but it's not that important as everything runs local anyways.